MainMenu Config Improvements

YoNeXia opened the pull request:
**Improve the MainMenu Interface for the facilitator.** Correspond to get the improvements of the issue #5 .

- Added instructions in left part, and change fields on right part.
- Added a message on the left of start button when bad information is completed in input fields.
- Now, check email field, adapt user and test field.

_Some options was commented in code, but it was because issue #6 talk about check if right informations are sended in DB. For the interface, do not need it._

**The new interface use layout to adapt at all screens.** But some times, if you have not the same size of the mockup, the interface are not totally correspond to the mockup.

YoNeXia committed:
Add me as contributor in README file

Add my name in README file, to see who contributed at this project.

YoNeXia committed:
Update MainScene to match with mockup

Change the layout of the interface to correspond with 16:9 format and the mockup.
Add some files to improve the layout (picts).

YoNeXia committed:
Add copyright to scripts files

Add copyright in the header of script files. This header is commented.

bastianilso reviewed (changes requested):
Congratulations with your first pull request, Yohann! 🐶 

Now, for some feedback.

Commit: **(update) change main menu interface options and add instructions**
- I think your commit titles are useful while you work on the code, but when we merge it into the main branch, I would like them to match the conventions used there. Please start the titles with a capital letter and please provide a body (smaller paragraph), that describes what changed (you can even mention what issue it is related to, ie this is related to Issue #5 ). See the examples [Chris Beams article](https://chris.beams.io/posts/git-commit/). Or see, for example [this commit](https://gitlab.gnome.org/GNOME/gnome-shell/commit/6205d5eb27dbab83f7387cb479cd3b5b7bd4495a).
- I have some comments on your code here, but let's first get the commits aligned properly. For now, let's keep this commit as the commit which changes the main menu interface.

Commit: **(add) copyright into new and modified files**
- Same comment on commit title/body.
- This commit should only add copyright information, but it also includes some changes to `MainMenu.unity`. I suggest you separate `MainMenu.unity` changes out into its own commit and squash it with "change main menu interface options and add instructions".

Commit: **(fix) change gameobjectname**
- Same comment on commit title/body.
- Given that the change is so minor, let's also squash this into "change main menu interface options and add instructions".

Commit: **(add) add me as contributor in README file**
- Same comment on commit title (this is simple enough that it will not need a body).
- Otherwise, looks good to me.

The other commits you will receive comment on, once we initially fix these commits. 👍 
If you want, I can show you how to make changes to your old commits, using a [Git Interactive Rebase](https://www.youtube.com/watch?v=ElRzTuYln0M) - Quentin might also be able to help you. This is an operation you can always abort, so dont worry if you mess up (your work is now on the Github server and can always be pulled down).

YoNeXia commented:
Hi Bastian!

I finished to do all modifications.
I changed commit names and descriptions. And I applied squash modifications.

You have to tell me about comments on my code at this commit 8d9297d. I am waiting your answer.
Futhermore, I am waiting your verification about commit changements.

Thanks 😉

bastianilso commented on Assets/Scripts/Menu/DestroyBlocker.cs:
```diff
@@ -0,0 +1,12 @@
+
+	// Update is called once per frame
+	void Update () {
+        Dropdown.Destroy(GameObject.Find("Blocker"));
```
Please see Unity's API for [GameObject.Find](https://docs.unity3d.com/ScriptReference/GameObject.Find.html)
"_For performance reasons, it is recommended to not use this function every frame._"

Just from looking at the code it is not clear what it does - what is a Blocker? Why do we want to destroy it on every frame update?

YoNeXia replied:
I delete this script because finally, just need to desactivate width force expand of the horizontal layout in the canvas.
So, as I modified MainScene.unity, I made a new commit 4d018a6.

bastianilso commented on Assets/Scripts/Menu/HighlightText.cs:
```diff
@@ -0,0 +1,32 @@
+    public void MouseInInputField()
+    {
+        // Set text in iput Field
+        text.text = "EDIT";
```
Let's just use a first-letter capitalized "Edit" here, rather than all-letter capitalized.

YoNeXia replied:
Fine, I modified it! :ok_hand:

bastianilso reviewed (changes requested):
@YoNeXia great! I had a look.

It's good that you split your work into several commits. But let's further squash your changes so they are a bit more clear to read. My suggestion is that we boil this to 7 commits (see suggestion at bottom).

But first, I will give you some new feedback related to commit messages:

Commit: **Change main menu interface options and add instructions**
- I would like if your commit message includes explaining that A) the commit introduces new assets required for the main menu design in issue #10 , B) what the purpose of DestroyBlocker and HighlightText is (what they are used for, why you are introducing them).
- I added some code change requests, have a look at the very bottom of this message.

Commit: **Add me as contributor in README file**
- Looks good to me now, there is a commit msg spelling mistake (s/contribuate/contribute) but that's it.

Commit: **Add text part to check empty or bad field**
- I would prepend this here with "MainMenu:" so the title says "MainMenu: Add text part to check empty or bad field." And perhaps write that you are adding a class, ie. "MainMenu: Add MissingField class to check for empty fields".
- This commit is making changes to all lines of code in LoggingManager, (removing everything and adding it again?). This might be caused by your IDE? If it is, I suggest that you drop those changes.
- The code in MissingField.cs looks fine - But the commit only adds the class, it doesnt actually make the program check for empty fields. The general convention we use is that our commits should represent additions of functionality - you might want to squash this commit together with whichever commit actually makes use of MissingField.cs. 


I will give you more feedback on your actual code, but first I want this boiled down to 7 commits, so it is easier for me to have a clear overview of what you did :bear:. This will require you to do some squashing, re-ordering etc in Git Interactive Rebase. Based on what I have read in your commit messages, my suggestion is the following:

1. a commit where you did your first main menu change (like the first commit is now).

1. a commit where you add yourself as contributor in the README file. (like the second commit)

1. a commit where you add functionality for checking for bad/empty fields (squash of relevant commits into third commit).

1. a commit where you took "commented out" unused UI from LoggingManager - and instead you now log some default values.

1. a commit where you add text to empty fields in the menu, ("Enter an user ID.") etc.

1. a commit where you make final adjustments to interface elements, to match the 16:9. This is a squash of "Modify interface", "Layout updates before verification, and possible improvements", "Eye part layout updated", "Add script to control font size", "Up part in instructions layout updated", "Font sizes in left part updated", "Instructions layout updated", "Last update for layout before verification", "Resize start button and change script to set user and test values", "Update font size and men picture size to correspond at all screen", "Reactivate white background in left part" and "Modify text in start button".

1. a commit where you add copyright info to all the files. a squash of "Add copyright into new and modified files" and "Add copyright to scripts files", etc.

YoNeXia commented:
> * This commit is making changes to all lines of code in LoggingManager, (removing everything and adding it again?). This might be caused by your IDE? If it is, I suggest that you drop those changes.

It is because I use automatic indentation in Visual Studio, and in this file, VS re-indent all the code. What I do?

YoNeXia commented:
> * I would like if your commit message includes explaining that A) the commit introduces new assets required for the main menu design in issue #10 , B) what the purpose of DestroyBlocker and HighlightText is (what they are used for, why you are introducing them).

I add this lines in the commit description :
- Add all new assets and textures for the new design (picts and font-style)
- Add scripts to control menu fields and dropdown. HighlightText.cs permits change the default value of any field by "Edit" when your mouse is inside. DestroyBlocker.cs is to destroy the blocker generate by the dropdown and this blocker creates a bug because it changes the size of dropdown when you open it.

YoNeXia committed:
Implement new main menu interface

Replace old interface by a new menu.
Left part contains instructions and right part contains the new menu.
Add all new assets and textures for the new design (picts and font-style)

YoNeXia committed:
MainMenu: Add script to create edit hover effect

HighlightText.cs permits change the default value of any field by "Edit" when your mouse is inside, and value entered when mouse is outside.
Add an header in new scripts. This header is in comment and contains copyright informations.

YoNeXia committed:
MainMenu: Add placeholder text for each field in the interface

Complete empty fields in Unity with comments to help the facilitator.

YoNeXia committed:
Add script to control font-size

Add script to set the same size of font of all instructions part.

YoNeXia committed:
Change linting in LoggingManager

Re-indent all the LoggingManager.cs with automatic command in Visual Studio.
Change all names of data to match with names in Unity.

YoNeXia committed:
Add script to check empty field.

Add new script to control text on left part of start button to inform the facilitator if he forget to complete any field.
A swicth case control if the correct error message is wrote into the interface when function is called.
LoggingManager.cs script updated to call function when empty field is detected.

YoNeXia committed:
Add email check with regex

Add a regex condition to check if a correct email is used.
Rename script to match with his function.

YoNeXia committed:
Check int values in some fields

Rename script HighlightText.cs to AdaptField.cs.
Add function in AdaptField.cs to check if int value entered in Participant and Test fields.
Add case in ErrorField.cs if not int value.

YoNeXia commented:
Hey @bastianilso ! 🙉

I finish to change all commits to match with your rules. Can you check it and give me a feed back?

Thanks. 👌

bastianilso reviewed (changes requested):
Commit **Change main menu interface**
- Looks good to me. Commit title could be slightly more clear, ie. "Implement new main menu interface", but that's all.

Commit **Add me as contributor in README file**
- Looks good to me. 👍 

Commit **Add script to control menu fields**
- Commit title could be more clear: Fx "MainMenu: Add script to create edit hover effect"
- Normally I would in a commit like this, also include a change to a *.unity file, since this would be necessary, in order for a reviewer to actually test your changes. But that's not easy to do retrospectively, so let's skip it this time around and just keep in mind for next pull request.

Commit **Update script LoggingManager.cs**
- Commit title is very generic. When changes don't change behavior, but just code style, you can refer to the changes as "linting changes". Fx "Change Linting in LoggingManager".
- Otherwise, looks fine. 

Commit **Add script to check empty field.**
- Looks good to me 🐤 

Commit **Add email check with regex**
- Looks good to me 🦆 

Commit **Comment all old parameters, toggles, not see in new interface**
- Instead of commenting out the code, I would like you to make the code use what corresponds to our default values. So, if ShowGazeDot by default is true, replace `private Toggle showGazeDot;` with `private bool showGazeDot = true;` and fix the code references accordingly. We still intend LoggingManager to log these values.
- This commit adds a `Debug.Log(switchInputDropdown.value);` which I don't think we need to commit.
- Commit title should specify that the changes relate to LoggingManager.

Commit **Add automatic comments in each field in interface**
- Code changes looks good to me.
- Commit title could be more clear. We would normally call this "placeholder text", a more accurate commit title would be: MainMenu: "Add placeholder text for each field in the interface."

Commit **Update MainScene to match with mockup**
- Looks good to me. 🐦 

Commit **Add script to control font-size**
- Why is it necessary to update the font size every frame on Update? This is both a question to you and also a request to have an explanation in the commit body. (Looks a bit like a hack 🔨 )
- Same comment as before - I would expect a change to a unity file here, so the change becomes testable, when I open the project.

Commit **Add copyright to scripts files**
- Looks good to me. 🐔 

Commit **Check int values in some fields**
- I find the implementation a bit unconventional here. My suggestion would be that you change `HighlightField` into a `MainMenuManager`, which keeps `string email`, `int participantID`, `int testNumber`, and `string comment`. In the `Start()` function you give them their default values and you change the UI placeholder text fields, ie. `participantField.text = Participant + participantID`. When users click the field, they update the conditionValue text, fx to "2". In MainMenuManager, you subscribe to the InputField `OnValueChanged()` event to capture when this changes and you use `Int.Parse()` to parse it. We can then get rid of the Regex `CheckIntValue()` function in LoggingManager - which really should only care about logging the values.

I am getting some spurious errors from Unity when trying your branch at the moment. But as long as you don't, I assume the issues are local on my end.

YoNeXia committed:
LoggingManager: omment all old parameters, toggles

Comment all datas not configured directly in new interface.

YoNeXia committed:
Fix email check

Change the method to check email. Is always a Regex method, but email check every inputfield end edit.
Method to check email is now in AdaptField.cs.
MainScene updated to add the event to check email in end edit of the inputfield.

YoNeXia committed:
Fix checkint method

Change place of CheckInt method from LoggingManager.cs to AdaptField.cs.
This method permit check if int value is entered every time when the value in int field is changed.

YoNeXia committed:
Update errors

Improve and simplify code to call an error and print it in UI.
Change name of variable to correspont with is function in ErrorField.cs.

YoNeXia committed:
Update font size and add text mesh pro

Add text mesh pro to control more text fields in instruction part. They permit to keep two headset lines in every size of the UI.

YoNeXia committed:
Fix int fields

Copy and adapt the implementation from the projet Whack A Mole, to simplify the code and understand in each project how the code work.
Now it is a button desactivate when you click on, and you can edit a int field, and the button display the value of the fied.

YoNeXia commented:
OK!
I change all bugs for the font size, I add the same implementation of Whack A Mole project for int fields, and I rebase all commits to match with your last message.

So please, now it's good 🙏

It is the picture of the mockup and UI in Unity to compare them.
![1234](https://user-images.githubusercontent.com/23256059/67642384-7ab2b780-f90b-11e9-94d0-858a272d30bc.png)

bastianilso commented:
:+1: good job @YoNeXia I have no further remarks. In case we find bugs or things need to change, we can revisit it later.
